src/cloud-resources/src/lib.rs

```diff
@@ -155,3 +155,15 @@ pub fn vpc_endpoint_name(id: GlobalId) -> String {
     // cloud infrastructure layer.
     format!("connection-{id}")
 }
+
+/// Returns the host to use for the VPC endpoint with the given ID and
+/// optionally in the given availability zone.
+pub fn vpc_endpoint_host(id: GlobalId, availability_zone: Option<String>) -> String {
```

Comment:
Can avoid the clone here by taking an `&str`, since `format!` will wind up allocating for the final string anyway.

```suggestion
pub fn vpc_endpoint_host(id: GlobalId, availability_zone: Option<&str>) -> String {
```

Reply:
Hmm but `aws_privatelink.availability_zone` is of type `Option<String>`, so I'd still need to map from `Option<String>` -> `Option<&str>`?

Reply:
Right! you can do `aws_privatelink.availability_zone.as_ref()` at the call site.

Reply:
Ah turns out it's `as_deref` in this case

Reply:
Ah right!